state/executor.go

```diff
@@ -129,6 +129,9 @@ func (e *BlockExecutor) CreateBlock(height uint64, lastCommit *types.Commit, las
 	block.SignedHeader.Header.LastCommitHash = e.getLastCommitHash(lastCommit, &block.SignedHeader.Header)
 	block.SignedHeader.Header.LastHeaderHash = lastHeaderHash
 	block.SignedHeader.Header.AggregatorsHash = state.Validators.Hash()
+	if state.NextValidators != nil {
+		block.SignedHeader.Header.NextAggregatorsHash = state.NextValidators.Hash()
```

Comment:
also, you need the updated state (which can be obtained after `ApplyBlock`) to capture any validator updates in the end block into your next validators. otherwise, the next validators might change after the `ApplyBlock` if there were any validator updates. it is bit tricky as `ApplyBlock` validates the block signature and also executes the block txns via proxyApp. you might need some refactoring of the `ApplyBlock` to make this work.

Reply:
Done.